plugins/techdocs-backend/src/service/router.ts

```diff
@@ -226,6 +242,23 @@ export async function createRouter(
     );
   });
 
+  // Ensures that the related entity exists and the current user has permission to view it.
+  router.use('/static/docs/:namespace/:kind/:name', async (req, _res, next) => {
+    const { kind, namespace, name } = req.params;
+    const entityName = { kind, namespace, name };
+    const token = getBearerToken(req.headers.authorization);
+
+    const entity = await entityLoader.load(entityName, token);
+
+    if (!entity) {
+      throw new NotFoundError(
+        `Entity not found for ${stringifyEntityRef(entityName)}`,
+      );
+    }
+
+    next();
+  });
```

Comment:
I'm still thinking of the edge(-ish) case where folks have custom object storage schemas in place (supported by custom TechDocsStorage API implementations in the front-end).  Is there an existing mechanism to short-circuit this in such a case?  (e.g. via some `app-config.yaml` value, etc)

Reply:
There's nothing in place at the moment. I was originally thinking that users could connect `Publisher.docsRouter()` to a different path, like `/static/custom-docs`, but then they can't take advantage of the techdocs cache. It's also easy enough to toggle the behavior with a config value if that's preferred though!

Reply:
Yeah, I'm especially curious on this because we as users of the plugin happen to fall into this category 😬 .

I was hoping there'd be some existing global-ish config that indicates "authz is explicitly enabled" and we can reliably, conditionally add the middleware in its presence?  Or is a TechDocs-specific exclude the better option?

Reply:
There is a `permission.enabled` config value, but that applies to the authorization framework as a whole. I wonder if this should somehow be encoded in the `Publisher` interface. Serving "non-catalog" techdocs requires a custom publisher implementation anyway, so having a way to determine this might be better / less redundant than a new config field. Something like `techdocs.authorizeStaticAssets = false` is certainly lower friction to implement though 😄

Reply:
> Edit: Perhaps we can look at the config value for techdocs.publisher.type and enable the behavior when it's one of the built-in options?

I'm not sure the type is going to be a reliable indicator; it controls the canonical data store and configs but not necessarily anything else. For instance, we use the vanilla GCS publisher for serving static assets, we just happen to have an entirely separate process for actually _publishing_ to the GCS bucket (external to backstage completely) at the moment.

Maybe a boolean flag like you mentioned is the only option? But maybe it'd be useful to share one config for all plugins? Not sure.

Reply:
What about a `publicDocsPath` config which this middleware can check to skip the catalog call? So if `publicDocsPath` matches the request path after `/static/docs`, then call `next()` and return early. The downside is that users will probably need to move things around in storage, but I like that it can work without disable techdocs authorization entirely

Reply:
I'll also add a check here so we only add the middleware if `permission.enabled = true`, which seems like a good optimization regardless of the custom docs setup

Reply:
> What about a publicDocsPath config which this middleware can check to skip the catalog call?

This is super interesting! Could even be useful in a default setup (eg make all docs under a namespace public?) but maybe that's complicated with the other metadata calls. Probably best to hold off until use-cases become clearer.

I'd be good with just the conditional middleware include on `permissions.enabled`.

Reply:
> Could even be useful in a default setup (eg make all docs under a namespace public?)

Smart! I like that idea, but you're right we should probably think more these use cases before introducing some "public docs" concept. I'll make a note to revisit this after.

> I'd be good with just the conditional middleware include on permissions.enabled.

Done in the latest commit